### PR TITLE
Public-friendly Users Find API

### PR DESCRIPTION
**Pull request by harimohanraj89**
**Story card:** https://www.pivotaltracker.com/story/show/171874600

## Because

The `GET /api/v3/users/find` endpoint returns too much PII about users on an unauthenticated endpoint.

## This addresses

* Addition of a `POST /api/v4/users/find` endpoint that finds by phone number only, and returns `id` and `sync_approval_status`. This information is not sensitive, and powers the first step of the sign up/in process on a new device.
* Correction of a few mistakes in the v3 API documentation
* Addition of some missing V3 specs

Full context can be found in [this PRD](https://docs.google.com/document/d/16-5-7yr62Hbzl2Yj8teEnb6LoUKcDsHCmps8ThnKu9s/edit?pli=1).

### CODE REVIEW

File: app/controllers/api/v4/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  skip_before_action :validate_sync_approval_status_allowed, only: [:find]
+  skip_before_action :authenticate, only: [:find]
+  skip_before_action :validate_facility, only: [:find]
+  skip_before_action :validate_current_facility_belongs_to_users_facility_group, only: [:find]
+
+  def find
+    return head :bad_request unless params[:phone_number].present?
+    user = PhoneNumberAuthentication.find_by(phone_number: params[:phone_number])&.user
+    return head :not_found unless user.present?
+    render json: { user: to_find_response(user) }, status: 200
```
Comment (vkrmis) [inline app/controllers/api/v4/users_controller.rb]: Is there a reason to prefer `status: 200` over `status: :ok`?

Comment (harimohanraj89) [reply app/controllers/api/v4/users_controller.rb]: Nope, they're the same. I notice that `200` is inconsistent. I'll update to use `status: :ok` 👍

File: spec/transformers/api/v3/user_transformer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        'otp',
+        'otp_valid_until',
+        'access_token',
+        'logged_in_at',
+        'role',
+        'organization_id'
+      )
+    end
+  end
+end
```
Comment (vkrmis) [inline spec/transformers/api/v3/user_transformer_spec.rb]: https://github.com/simpledotorg/simple-server/pull/858/commits/7445bc8e976846b37cfca2b68b58c1fa2050324d#r397756160

File: spec/transformers/api/v4/user_transformer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        'sync_approval_status' => 'approved'
+      )
+    end
+
+    it 'excludes other params' do
+      expect(response).not_to include('other', 'field')
+    end
+
+    it 'excludes sensitive params' do
+      expect(response).not_to include('otp',
```
Comment (kitallis) [inline spec/transformers/api/v4/user_transformer_spec.rb]: Should `password_digest` be also in that list or is that never possible anymore?

Comment (harimohanraj89) [reply spec/transformers/api/v4/user_transformer_spec.rb]: Makes sense to include it in this sensitive list 👍

Comment (harimohanraj89) [reply spec/transformers/api/v4/user_transformer_spec.rb]: Done 👍

File: spec/transformers/api/v3/user_transformer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+RSpec.describe Api::V3::UserTransformer do
+  describe 'to_response' do
+    let!(:user) { create(:user) }
+
+    subject(:response) { described_class.to_response(user) }
+
+    before do
+      allow(Api::V3::Transformer).to receive(:to_response)
+        .with(user)
+        .and_return({ 'user' => 'attributes'})
```
Comment (harimohanraj89) [inline spec/transformers/api/v3/user_transformer_spec.rb]: Add sensitive attributes to this stubbed response to make the assertion below meaningful.

File: app/schema/api/v4/models.rb [hunk 5]
```diff
@@ -82,6 +91,7 @@ def activate_user
         required: %w[id password] }
     end
 
+
```
Comment (harimohanraj89) [inline app/schema/api/v4/models.rb]: 👾

File: swagger/v2/swagger.json [hunk 6]
```diff
@@ -1874,6 +1865,15 @@
...
           ]
+        },
+        "phone_number": {
+          "$ref": "#/definitions/non_empty_string"
+        },
+        "password_digest": {
+          "$ref": "#/definitions/bcrypt_password"
+        },
+        "registration_facility_id": {
+          "$ref": "#/definitions/uuid"
```
Comment (harimohanraj89) [inline swagger/v2/swagger.json]: Ignore this diff

File: swagger/v3/swagger.json [hunk 7]
```diff
@@ -2011,6 +2002,15 @@
...
           ]
+        },
+        "phone_number": {
+          "$ref": "#/definitions/non_empty_string"
+        },
+        "password_digest": {
+          "$ref": "#/definitions/bcrypt_password"
+        },
+        "registration_facility_id": {
+          "$ref": "#/definitions/uuid"
```
Comment (harimohanraj89) [inline swagger/v3/swagger.json]: Ignore this diff

**Pull request merged**